harness/determined/estimator/_estimator_trial.py

```diff
@@ -504,13 +522,20 @@ def _init_model(self) -> None:
         self.train_spec = tf.estimator.TrainSpec(
             input_fn=repeating_train_fn, hooks=self.train_hooks
         )
+        steps = (
+            self.val_spec.steps // self.context.distributed.get_size()
+            if self.val_spec.steps is not None
+            else None
+        )
         self.eval_spec = tf.estimator.EvalSpec(
-            input_fn=self.val_spec.input_fn, hooks=self.val_spec.hooks, steps=None
+            input_fn=self.val_spec.input_fn, hooks=self.val_hooks, steps=steps
```

Comment:
non-blocking: just to reduce number of class members could we do:

```
self.eval_spec = tf.estimator.EvalSpec(
      input_fn=self.val_spec.input_fn, hooks=self. _init_val_hooks(), steps=steps
```

I would probably do the same for train hooks.

Reply:
I'll do this as a separate PR